File: KmmTestApp/composeApp/src/commonMain/kotlin/com/gipstech/kmmtestapp/MyCalibrationListener.kt
package com.gipstech.kmmtestapp

import com.gipstech.mdk.CalibrationListener
import com.gipstech.mdk.CalibrationManager
import com.gipstech.mdk.LocationListener

class MyCalibrationListener(private val locationListener: LocationListener): CalibrationListener {
    override suspend fun onProgress(percentage: Int, enough: Boolean) {
        LocationManager.locationMessage.value = "Calibration: $percentage%"
        if (enough) {
            LocationManager.locationMessage.value = "Calibration done!"
            CalibrationManager.endCalibration()
            LocationManager.place?.startLocalization(locationListener)
        }
    }
}

File: SwiftTestApp/SwiftTestApp/MyCalibrationListener.swift
//
//  MyCalibrationListener.swift
//  SwiftTestApp
//
//  Created by GiPStech on 25/03/25.
//

import GiPStechMDK

class MyCalibrationListener: CalibrationListener {
    var manager: LocationManager
    
    init(manager: LocationManager) {
        self.manager = manager
    }
    
    func __onProgress(percentage: Int32, enough: Bool) async throws {
        manager.report("Calibration \(percentage)%")
        if enough {
            manager.report("Calibration done!")
            try await CalibrationManager.shared.endCalibration()
            try await manager.restartLocalization()
        }
    }
}
